rpcs3/Emu/Cell/lv2/sys_timer.cpp

```diff
@@ -93,6 +92,8 @@ error_code sys_timer_destroy(u32 timer_id)
 			return CELL_EISCONN;
 		}
 
+		// Signal thread using invalid state
+		timer.state = -1;
```

Comment:
What race?

Reply:
state setting can be replaced with STOP instead of -1.
To solve this I've moved timer.state store to lock scope, and checked the state in the timer thread's lock scope.

Reply:
another way to solve it is waiting for the thread to finnish current time check in sys_timer_stop,

Reply:
Correct fix is compare_and_swap in timer thread, not removing on_abort.